javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1392,24 +1392,37 @@ public T getNext() {
             final Iterator<T> that = this;
             return new AbstractIterator<T>() {
 
-                private T next = null;
+                private T next;
```

Comment:
From what I can see the `concat(this)` will not work. We need to concat the remaining elements _after_ t.